<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Lemon</class>
 <widget class="QMainWindow" name="Lemon">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>725</width>
    <height>510</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>725</width>
    <height>510</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Lemon</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resource.qrc">
    <normaloff>:/icon/icon.png</normaloff>:/icon/icon.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="styleSheet">
       <string notr="true">font-size: 9pt;</string>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tasksTab">
       <attribute name="title">
        <string>Tasks</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QGroupBox" name="taskBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
            <horstretch>1</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>176</width>
            <height>387</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">font-size: 9pt;</string>
          </property>
          <property name="title">
           <string>Summary</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="SummaryTree" name="summary">
             <property name="contextMenuPolicy">
              <enum>Qt::DefaultContextMenu</enum>
             </property>
             <property name="styleSheet">
              <string notr="true">font-size: 10pt;</string>
             </property>
             <attribute name="headerVisible">
              <bool>false</bool>
             </attribute>
             <column>
              <property name="text">
               <string notr="true">1</string>
              </property>
             </column>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="editBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
            <horstretch>3</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="styleSheet">
           <string notr="true">font-size: 9pt;</string>
          </property>
          <property name="title">
           <string>Detail</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <item>
            <widget class="QStackedWidget" name="editWidget">
             <property name="contextMenuPolicy">
              <enum>Qt::NoContextMenu</enum>
             </property>
             <property name="currentIndex">
              <number>2</number>
             </property>
             <widget class="QWidget" name="nullPage"/>
             <widget class="TaskEditWidget" name="taskEdit"/>
             <widget class="TestCaseEditWidget" name="testCaseEdit"/>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="contestantsTab">
       <attribute name="title">
        <string>Contestants</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="ResultViewer" name="resultViewer">
          <property name="styleSheet">
           <string notr="true">font-size:9pt;</string>
          </property>
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <property name="tabKeyNavigation">
           <bool>false</bool>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::ExtendedSelection</enum>
          </property>
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectRows</enum>
          </property>
          <property name="verticalScrollMode">
           <enum>QAbstractItemView::ScrollPerPixel</enum>
          </property>
          <property name="horizontalScrollMode">
           <enum>QAbstractItemView::ScrollPerPixel</enum>
          </property>
          <property name="showGrid">
           <bool>false</bool>
          </property>
          <property name="sortingEnabled">
           <bool>true</bool>
          </property>
          <property name="cornerButtonEnabled">
           <bool>false</bool>
          </property>
          <attribute name="horizontalHeaderHighlightSections">
           <bool>false</bool>
          </attribute>
          <attribute name="verticalHeaderVisible">
           <bool>false</bool>
          </attribute>
          <attribute name="verticalHeaderDefaultSectionSize">
           <number>25</number>
          </attribute>
          <attribute name="verticalHeaderMinimumSectionSize">
           <number>25</number>
          </attribute>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>468</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="refreshButton">
            <property name="text">
             <string>&amp;Refresh</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="judgeButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>&amp;Judge</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="judgeAllButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Judge &amp;All</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>725</width>
     <height>23</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">font-size:9pt;</string>
   </property>
   <widget class="QMenu" name="fileMenu">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="newAction"/>
    <addaction name="openAction"/>
    <addaction name="closeAction"/>
    <addaction name="separator"/>
    <addaction name="exitAction"/>
   </widget>
   <widget class="QMenu" name="controlMenu">
    <property name="title">
     <string>&amp;Control</string>
    </property>
    <addaction name="judgeAction"/>
    <addaction name="judgeAllAction"/>
    <addaction name="separator"/>
    <addaction name="addTasksAction"/>
    <addaction name="makeSelfTestAction"/>
    <addaction name="exportAction"/>
   </widget>
   <widget class="QMenu" name="toolsMenu">
    <property name="title">
     <string>&amp;Tools</string>
    </property>
    <widget class="QMenu" name="languageMenu">
     <property name="title">
      <string>UI Language</string>
     </property>
     <addaction name="setEnglishAction"/>
    </widget>
    <addaction name="optionsAction"/>
    <addaction name="languageMenu"/>
   </widget>
   <widget class="QMenu" name="helpMenu">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="aboutAction"/>
   </widget>
   <addaction name="fileMenu"/>
   <addaction name="controlMenu"/>
   <addaction name="toolsMenu"/>
   <addaction name="helpMenu"/>
  </widget>
  <action name="newAction">
   <property name="text">
    <string>&amp;New Contest</string>
   </property>
  </action>
  <action name="openAction">
   <property name="text">
    <string>&amp;Open Existing Contest</string>
   </property>
  </action>
  <action name="exitAction">
   <property name="text">
    <string>E&amp;xit</string>
   </property>
  </action>
  <action name="optionsAction">
   <property name="text">
    <string>&amp;Options</string>
   </property>
  </action>
  <action name="aboutAction">
   <property name="text">
    <string>&amp;About</string>
   </property>
  </action>
  <action name="judgeAction">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Judge Selected</string>
   </property>
  </action>
  <action name="judgeAllAction">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Judge &amp;All</string>
   </property>
  </action>
  <action name="closeAction">
   <property name="text">
    <string>&amp;Close Current Contest</string>
   </property>
  </action>
  <action name="setEnglishAction">
   <property name="text">
    <string notr="true">English</string>
   </property>
  </action>
  <action name="addTasksAction">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Add &amp;Tasks Automatically</string>
   </property>
  </action>
  <action name="makeSelfTestAction">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Make Self-testing Folder</string>
   </property>
  </action>
  <action name="exportAction">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Export Result</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>SummaryTree</class>
   <extends>QTreeWidget</extends>
   <header>summarytree.h</header>
  </customwidget>
  <customwidget>
   <class>TaskEditWidget</class>
   <extends>QWidget</extends>
   <header>taskeditwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TestCaseEditWidget</class>
   <extends>QWidget</extends>
   <header>testcaseeditwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ResultViewer</class>
   <extends>QTableWidget</extends>
   <header>resultviewer.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>summary</tabstop>
  <tabstop>refreshButton</tabstop>
  <tabstop>judgeButton</tabstop>
  <tabstop>judgeAllButton</tabstop>
  <tabstop>resultViewer</tabstop>
 </tabstops>
 <resources>
  <include location="resource.qrc"/>
 </resources>
 <connections/>
</ui>
